src/connectivity.jl

```diff
@@ -728,17 +728,29 @@ According to Erdös-Gallai theorem, a degree sequence ``\\{d_1, ...,d_n\\}`` (so
 ```math
 \\sum_{i=1}^{r} d_i \\leq r(r-1) + \\sum_{i=r+1}^n min(r,d_i)
 ```
-for each integer r <= n-1
+for each integer r <= n-1. 
+
+See also: [`isdigraphical`](@ref)
 """
-function isgraphical(degs::Vector{<:Integer})
+function isgraphical(degs::AbstractVector{<:Integer})
+    # Check whether the degree sequence is empty
+    !isempty(degs) || return true
+    # Check whether the degree sequence has only non-negative values
+    all(degs .>= 0) ||
+        throw(ArgumentError("The degree sequence must contain non-negative integers only."))
```

Comment:
Wouldn't it be more reasonable to just return `false` here? Of course if that is so, we should probably do the same thing for `isdigraphical`.

Reply:
We thought that `false` was reserved for when "the degree sequence is valid but not graphical", while errors should be thrown if the degree sequence is not even valid.

If it isn't appropriate we will change the behavior.

Reply:
I leave the decision up too you. One thing that is nice about not throwing an error, that a user does not have to make an extra check for an error, if their input might contain a negative number.

But then we probably should also check, that all degrees are less or equal than `n - 1`?

Where is would still leave the throws check in, is in the `isdigraphical` method for inputs of different lengths.

Also note that by throwing an exception in `isgraphical` we kind of introduce a breaking change - but as this is a very rare edge case, I think that should be ok.

Let me know what you decide.

Reply:
Thanks for your argumentation. 

We agree that returning `false` allows for simpler usage, and if the user wishes to get an error in specific circumstances, she/he may wrap the function. Thus we now return `false` when negative degrees are present, or when a degree exceeds `n-1`.